services/skus/handler/handler_test.go

```diff
@@ -504,6 +515,129 @@ func TestOrder_CreateNew(t *testing.T) {
 	}
 }
 
+func TestOrder_Cancel(t *testing.T) {
+	type tcGiven struct {
+		ctx context.Context
+		svc *mockOrderService
+		oid uuid.UUID
+	}
+
+	type tcExpected struct {
+		err *handlers.AppError
+	}
+
+	type testCase struct {
+		name  string
+		given tcGiven
+		exp   tcExpected
+	}
+
+	tests := []testCase{
+		{
+			name: "invalid_orderID",
+			given: tcGiven{
+				ctx: context.WithValue(context.Background(), chi.RouteCtxKey, &chi.Context{
+					URLParams: chi.RouteParams{
+						Keys:   []string{"orderID"},
+						Values: []string{"invalid_id"},
+					},
+				}),
+				svc: &mockOrderService{},
+				oid: uuid.Nil,
+			},
+			exp: tcExpected{
+				err: handlers.ValidationError("request", map[string]interface{}{"orderID": "uuid: incorrect UUID length: invalid_id"}),
```

Comment:
I agree that it's tightly coupled to the library. Fixed by reporting to the outside with a sentinel error. The test has been updated to check for that as well.

> Assert on what we own such as error contains "orderID" or simply that we received a non nil validation error.
Disagree. Whenever it's possible to check a specific error, it makes sense to do so. So the test remains specific to the error. This is because we are in control of what's returned, hence we can also check that we in fact respond with what we expect.

Reply:
I am not sure what you are disagreeing with, the point to the original comment  is not asserting on the text returned by the satori library i.e. `"uuid: incorrect UUID length: invalid_id"`. Whether you address it by adding a sentinel error or asserting on some other value that is not returned by the library works. The only downside to using a sentinel is you do loose the original error with is now not logged opposed to having a slightly looser test.